### DISCUSSION TITLE
How can I connect different serie with a line? [URGENT]

### DISCUSSION THREAD

**Discussion by AkshaySharmaDEV** [XCharts-Team/XCharts#308] [General]
I have three different serie, and I want to connect all the series' points to see a line. I want something like this:
![line-graph](https://github.com/XCharts-Team/XCharts/assets/68364794/ed56bcdd-196b-4bc4-a96b-fe210c3c79a4)

The current graph only needs the line to be generated between these points:
![observation](https://github.com/XCharts-Team/XCharts/assets/68364794/9888ad71-e826-44f8-87d9-8562290bafda)

**Comment by Bian-Sh** [comment]
我觉得将不同的 Serial 练成线这个是一个不合理不科学的做法，就算这些离散的点连线了，这又有何现实意义呢？
你可以使用新的柱状图结合折线图描述新的观点和结论，而不是采取这种看着就让人费解的方式实现

I think it is unreasonable and unscientific to connect different serials into a line. Even if these discrete points are connected by lines, what practical significance does it have?
You can use a new combination of bar chart and line graph to describe new perspectives and conclusions, instead of implementing this confusing method.

**Comment by monitor1394** [comment]
If `Legend` is not displayed, `VisualMap` can implement line of different color segments.
To fully implement your need, you can draw lines in a custom drawing, see [Example12_CustomDraw.cs](https://github.com/XCharts-Team/XCharts/blob/master/Examples/Example12_CustomDrawing.cs):

```
var points = new List<Vector3>();
chart.onDraw = delegate (VertexHelper vh)
{
    points.Clear();
    foreach (var serie in chart.series)
    {
        var dataPoints = serie.context.dataPoints;
        for (int i = 0; i < dataPoints.Count; I++)
        {
            points.Add(dataPoints[I]);
        }
    }
    if (points.Count > 1)
    {
        var sp = points[0];
        for (int i = 1; i < points.Count; I++)
        {
            var ep = points[i];
            var color = chart.theme.GetColor(i - 1);
            UGL.DrawLine(vh, sp, ep, chart.theme.serie.lineWidth, color);
            sp = ep;
        }
    }
};
```

<img width="305" alt="image" src="https://github.com/XCharts-Team/XCharts/assets/1450329/a6f525f8-b5ea-452a-8fc4-cf042d86b878">
